packages/testing/config/storybook/main.js

```diff
@@ -136,6 +136,12 @@ const baseConfig = {
   ...(process.env.NODE_ENV !== 'production' && {
     staticDirs: [`${staticAssetsFolder}`],
   }),
+  // https://storybook.js.org/docs/react/configure/typescript#mainjs-configuration
+  typescript: {
+    check: true,
+    // By default, the checker runs asynchronously in dev mode. Force it to run synchronously.
+    checkOptions: { async: false },
+  },
```

Comment:
@virtuoushub unfortunately, this throws an error for JS projects (i.e. projects without a tsconfig.json), which is the case for our E2E CI as well as other projects. 

You could add a conditional to confirm project targets TS before running this. There's an example of utility over here for [Generate Command](https://github.com/redwoodjs/redwood/blob/83bd510dfade35076a2c941a957158a0634a1eb5/packages/cli/src/commands/generate.js), but it's probably overkill to import the entire structure package for this... 🤔